src/bin/cargo-flamegraph.rs

```diff
@@ -343,7 +344,10 @@ fn find_unique_target(
                     None => true,
                     Some(default_name) => &t.name == default_name,
                 };
-                (ok_kind && default_filter).then(|| BinaryTarget {
+                let name_filter = target_name
+                    .map(|target_name| t.name == target_name)
+                    .unwrap_or(true);
```

Comment:
I find the spelling like `default_filter` (with an explicit `match`) a good deal easier to follow than the `map().unwrap_or()` combinators. Would you mind rewriting it that way?